### PR TITLE
add dry_run mode, support deleting group registered runners

### PR DESCRIPTION
**Pull request by itsmethemojo**
i added following features to support my usescases
 
 * dry run mode (DRY_RUN)
 * optional basic auth credentials
 * optional giving in a group id to delete runners registered on the specific group (thats interesting if you own a group but not the gitlab)

### CODE REVIEW

File: cleanup_gitlab_runner.py [hunk 1]
```diff
@@ -77,8 +93,15 @@ def main():
             sys.stdout.write(f"skip runner {runner.description} (id: {runner.id}) because is online\n")
             continue
         try:
-            runner.delete()
+            if env_vars.dry_run:
+                sys.stdout.write(f"[DRY RUN] delete runner {runner.description} (id: {runner.id})\n")
+                continue
+            if env_vars.gitlab_group:
+                conn.runners.get(runner.id).delete()
+            else:
```
Comment (gi8lino) [inline cleanup_gitlab_runner.py]:
```suggestion
                continue
```

File: cleanup_gitlab_runner.py [hunk 2]
```diff
@@ -77,8 +93,15 @@ def main():
...
+                runner.delete()
```
Comment (gi8lino) [inline cleanup_gitlab_runner.py]:
```suggestion
            runner.delete()
```

**Review by gi8lino — CHANGES_REQUESTED** [review]
hi @itsmethemojo 
I don't like `else` 🙈
rest is great👍

**Comment by itsmethemojo** [comment]
~~should i manually squash or will this anyway be done with the PR?~~ squashed

**Comment by gi8lino** [comment]
Hi @itsmethemojo 
Thank you very much for your contribution!

**Pull request merged**